### PR TITLE
ID panel changes

### PR DESCRIPTION
**Pull request by Core0verload**
ID panel (the one used in labs and bunkers) now opens the locked doors instead of deleting them. It wouldn't anymore give "The nearby doors slide into the floor." message if the doors are already opened or destroyed.

Their radius was reduced too: 2 radius is enough for both bunker and lab, and 3 radius was breaking doors in my reworked bunker map (coming soon).

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -621,8 +616,18 @@ void iexamine::cardreader(player *p, map *m, const tripoint &examp)
             }
         }
         add_msg(_("You insert your ID card."));
-        add_msg(m_good, _("The nearby doors slide into the floor."));
         p->use_amount(card_type, 1);
+
+        int open = 0;
```
Comment (illi-kun) [inline src/iexamine.cpp]: why it's not a boolean value?

**Comment by nanovad** [comment]
If the doors slide into the floor, how do they become closeable?
This seems a little pedantic but I imagined metal doors as big side-hinged plates that can be swung open and closed like a regular house door.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -621,8 +616,18 @@ void iexamine::cardreader(player *p, map *m, const tripoint &examp)
             }
         }
         add_msg(_("You insert your ID card."));
-        add_msg(m_good, _("The nearby doors slide into the floor."));
         p->use_amount(card_type, 1);
+
+        bool open = 0;
```
Comment (BevapDin) [inline src/iexamine.cpp]: It's a `bool`, why are you assigning an `int`? Same below with `1`, why the `int`? It should be `true` / `false`.

**Comment by Coolthulhu** [comment]
The doors become closeable after sliding into the floor. As in, manually closeable. That would require the survivor to pull them out from the floor.

**Comment by Core0verload** [comment]
I changed the text, now the door just "slides open". It may mean sliding into a nearby wall as well.

**Comment by Coolthulhu** [comment]
They are still closeable by hand. This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

**Comment by nanovad** [comment]
I think this would require another door type to get the wanted functionality - or, perhaps the door could be "locked" and only openable with a card reader?

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a manual control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency.

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency, like in real life bunkers.

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency, like in real life bunkers.

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency, like in real life bunkers.

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency, like in real life bunkers.

**Comment by Core0verload** [comment]
>This shouldn't be possible after they slide open, especially not when the system that manipulates them is still in working order.

Maybe there is a control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency, like in real life bunkers.

**Comment by Cat09** [comment]
if this electronic door system is to be reclosable, then it needs to be activated by another control panel (with an ID) to do so. I can see there being a wall panel with a manual crank that works only when the power is out, though.

**Comment by Coolthulhu** [comment]
> Maybe there is a manual control button on the unlocked panel, and every time survivor closes the door, he pushes that button. Or there is a special manual mechanism to close or open the door in case of emergency. 

That button wouldn't allow closing just one tile of the double door.

**Comment by Coolthulhu** [comment]
If this one is to be merged, it needs some other way of designating the close-able doors. For example, a special floor tile that can be turned back into doors by using the card reader once again.

**Comment by mugling** [comment]
Appears stalled. Closed to reduce the PR queue. Could be reopened if OP wants to rebase and fix outstanding issues?

**Pull request closed without merging**